discount/src/main/java/com/kyriba/conference/discount/service/DiscountServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.kyriba.conference.discount.service;
+
+import com.kyriba.conference.discount.api.dto.DiscountExternalDto;
+import com.kyriba.conference.discount.api.dto.DiscountResponse;
+import com.kyriba.conference.discount.api.dto.DiscountUpdateParamsDto;
+import com.kyriba.conference.discount.dao.DiscountRepository;
+import com.kyriba.conference.discount.domain.DiscountEntity;
+import com.kyriba.conference.discount.domain.DiscountType;
+import com.kyriba.conference.discount.util.DiscountDataMapper;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * @author Igor Lizura
+ */
+@Service
+@Transactional
+public class DiscountServiceImpl implements DiscountService{
+    private final DiscountRepository discountRepository;
+
+    public DiscountServiceImpl(DiscountRepository discountRepository) {
+        this.discountRepository = discountRepository;
+    }
+
+    @Override
+    public List<DiscountExternalDto> getAllDiscounts() {
+        return discountRepository.findAll().stream().map(DiscountDataMapper.MAPPER::entityToDto).collect(Collectors.toList());
```

Comment:
mapping logic could be moved to entity level

Reply:
Mapping isn't a business rule, why it should be in a domain object?

Reply:
Mapping is converting internal state of domain model to transfer it outside. Incapsulation drives us to have such logic in domain entity itself.